hooks/frame_operations_tk-3dsmaxplus.py

```diff
@@ -0,0 +1,52 @@
+# Copyright (c) 2013 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import MaxPlus
+
+import sgtk
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+
+class FrameOperation(HookBaseClass):
+    """
+    Hook called to perform a frame operation with the
+    current scene
+    """
+
+    def get_frame_range(self, **kwargs):
+        """
+        get_frame_range will return a tuple of (in_frame, out_frame)
+
+        :returns: Returns the frame range in the form (in_frame, out_frame)
+        :rtype: tuple[int, int]
+        """
+        ticks = MaxPlus.Core.EvalMAXScript("ticksperframe").GetInt()
+        current_in = MaxPlus.Animation.GetAnimRange().Start() / ticks
+        current_out = MaxPlus.Animation.GetAnimRange().End() / ticks
+        return (current_in, current_out)
+
+    def set_frame_range(self, in_frame=None, out_frame=None, **kwargs):
+        """
+        set_frame_range will set the frame range using `in_frame` and `out_frame`
+
+        :param int in_frame: in_frame for the current context
+            (e.g. the current shot, current asset etc)
+
+        :param int out_frame: out_frame for the current context
+            (e.g. the current shot, current asset etc)
+
+        :returns: Returns if the operation was successfull
+        """
+
+        ticks = MaxPlus.Core.EvalMAXScript("ticksperframe").GetInt()
+        range = MaxPlus.Interval(in_frame * ticks, out_frame * ticks)
+        MaxPlus.Animation.SetRange(range)
+        return True
```

Comment:
See above comment - I think it should be ok to assume success unless an exception is raised from the hook, so this method shouldn't need to return anything.

Reply:
Done in all hooks